common/iovector.h

```diff
@@ -827,7 +828,7 @@ class iovector
     uint16_t capacity;              // total capacity
     uint16_t iov_begin, iov_end;    // [iov_begin, iov_end)
     uint16_t nbases;                // # of allocated pointers
-    struct iovec iovs[0];
```

Comment:
[0] is a common practice, and it's more read-able than iovs_ptr(). We can simple ignore the warning with ```#pragma``

Reply:
It is a common solution in GNU C extension.
C99 standard add new define of flex array, `struct iovec iovs[]` to present a flex array.
though, C++ does not allow inheritance by a class with tailed flex array, that is why here replaced by iovs_ptr()

Seems there is not a better solution than `iovs[0]`